libvirt/tests/src/migration/migration_with_disk/migration_with_fdgroup.py

```
@@ -0,0 +1,204 @@
+        cmd = (f'nohup sh -c \'(echo "dom-fd-associate {vm_name} {fdgroup_vdb} {fd_id_vdb}{flag}"; '
+               f'echo "dom-fd-associate {vm_name} {fdgroup_vdc} {fd_id_vdc}{flag}"; '
+               f'tail -f /dev/null) | virsh\' {fd_id_vdb}<>{file_vdb} {fd_id_vdc}<>{file_vdc} '
+               f'> /dev/null 2>&1 &')
+        test.log.debug("Running FD association on dest: %s", cmd)
+        remote.run_remote_cmd(cmd, params, runner, ignore_status=False)
+    else:
+        cmd = (f'virsh "dom-fd-associate {vm_name} {fdgroup_vdb} {fd_id_vdb}{flag}; '
+               f'dom-fd-associate {vm_name} {fdgroup_vdc} {fd_id_vdc}{flag}; '
+               f'start {vm_name}" {fd_id_vdb}<>{file_vdb} {fd_id_vdc}<>{file_vdc}')
+        test.log.debug("Running command on source: %s", cmd)
+        result = process.run(cmd, ignore_status=True, shell=True)
```

Ruff (0.15.6)

[error] Function call with `shell=True` parameter identified, security issue

(S604)